### PR TITLE
Address review comments from PR #7434

### LINKED ISSUE #7284: Implement custom fees charging step in `CryptoTransfer`
- [x] Fixed Fee
- [ ] Fractional Fee
- [ ] RoyaltyFee

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #7284 
 Addressed review comments from PR https://github.com/hashgraph/hedera-services/pull/7434 and added unit tests

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/AssessmentResult.java [hunk 1]
```diff
@@ -30,37 +30,43 @@
...
     /* And for each "assessable change" that can be charged a custom fee, delegate to our
     fee assessor to update the balance changes with the custom fee. */
     private List<AssessedCustomFee> assessedCustomFees;
 
     public AssessmentResult(
             final List<TokenTransferList> inputTokenTransfers, final List<AccountAmount> inputHbarTransfers) {
-        inputTokenAdjustments = buildFungibleTokenTransferMap(inputTokenTransfers);
+        mutableInputTokenAdjustments = buildFungibleTokenTransferMap(inputTokenTransfers);
+        immutableInputTokenAdjustments = mutableInputTokenAdjustments.entrySet().stream()
+                .collect(Collectors.collectingAndThen(
```
Comment (tinker-michaelj) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/AssessmentResult.java:53]: _Nit_ - the `Collector` factories `collectingAndThen()`, `reducing()`, `filtering()`, etc. are only meant to be used as "downstream" collectors in for a [`Collectors.groupingBy()`](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#groupingBy-java.util.function.Function-java.util.stream.Collector-) usage.

Within a normal stream, you should be able to perform the desired aggregation on the source stream, e.g.
```
immutableInputTokenAdjustments = Collections.unmodifiableMap(
    mutableInputTokenAdjustments.entrySet().stream()
        .collect(Collectors.toMap(Map.Entry::getKey, entry -> Map.copyOf(entry.getValue()))));
```

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/transfer/CustomFeeAssessmentStepTest.java [hunk 2]
```diff
@@ -335,17 +336,18 @@ void htsFixedFeeAndRoyaltyFeeWithFallbackNetOfTransfersSelfDenomination() {
                 fungibleTokenId,
```
Comment (tinker-michaelj) [inline hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/transfer/CustomFeeAssessmentStepTest.java:336]: Great tests here. 🤩

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @Neeharika-Sompalli ! 🥂

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/AssessmentResult.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/AssessmentResult.java:53]: yes, will make that change Michael. tyvm !

**Pull request merged**